### PR TITLE
Compute a build duration report

### PR DESCRIPTION
**Pull request by chouquette**
This will allow us to have a better observability on our build & package durations, as we currently don't have a way (other than manually reading all logs) to know what takes most of our build times

### CODE REVIEW

**Comment by chouquette** [comment]
test build: https://gitlab.ddbuild.io/DataDog/datadog-agent/-/pipelines/28524917

File: lib/omnibus/project.rb [hunk 1]
```diff
@@ -1402,6 +1403,15 @@ def built_manifest
...
     end
 
+    def build_summary
+      summary = {"build" => {}}
+      softwares.each do |s|
+        summary["build"][s.name] = s.build_summary
+      end
+      summary["packaging"] = @package_summary
+      summary
+    end
```
Comment (Pythyu) [inline lib/omnibus/project.rb]: 🤔 might be me being a ruby noob but I'm not finding where this function is called ?

**Review by Pythyu — APPROVED** [review]
Otherwise LGTM

File: lib/omnibus/project.rb (hunk 1 above)
Comment (chouquette) [reply lib/omnibus/project.rb]: Right there: https://github.com/DataDog/omnibus-ruby/pull/183/files#diff-07a7d7bffaf8905a505d0201c036661c95f08b2cb4d5d86c13f7fe774bcfe9e1R1472
I agree that it's not obvious at all, but it duplicates what's done for the catalog manifest

**Comment by chouquette** [comment]
Marking as "no auto bump" since datadog agent is frozen, and the bump will occur within the PR that uses the build summary

**Pull request merged**